src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final String EXEPTION_MASAGE = "Invalid index";
+    private T[] defaultArray;
+    private int size;
+
+    public ArrayList() {
+        defaultArray = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        capacityCheck();
+        defaultArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        capacityCheck();
+        if (isIndexValid(index) || index == size) {
+            for (int i = size; i > index; i--) {
+                defaultArray[i] = defaultArray[i - 1];
+            }
+            defaultArray[index] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException(EXEPTION_MASAGE);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        size += list.size();
+        int indexList = 0;
+        capacityCheck();
+        for (int i = size - list.size(); i < size; i++) {
+            defaultArray[i] = list.get(indexList);
+            indexList++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (isIndexValid(index)) {
```

Comment:
we can throw `ArrayListIndexOutOfBoundsException(EXEPTION_MASAGE);` inside `isIndexValid` method, this will reduce number of duplicated code